### PR TITLE
Refactor: Extract Gateway Donation Summary to framework.

### PR DESCRIPTION
**Pull request by kjohnson**
## Description

This PR extracts the Gateway Donation Summary functionality from the Stripe Gateway update.

## Pre-review Checklist

-   [ ] Acceptance criteria satisfied and marked in related issue
-   [x] Relevant `@unreleased` tags included in DocBlocks
-   [x] Includes unit and/or end-to-end tests
-   [x] [Self Review](https://give.gitbook.io/development-manual/devops/github/code-reviews#self-review) of code and UX completed

### CODE REVIEW

**Comment by ravinderk** [comment]
@kjohnson how I will be able to integrate recurring donations with it?
Did you think about this?

**Review by ravinderk — APPROVED** [review]

File: src/Framework/PaymentGateways/DonationSummary.php [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    }
+
+    /**
+     * @unreleased
+     * @return string
+     */
+    protected function getLabel()
+    {
+        $formId = give_get_payment_form_id( $this->get( 'donationId' ) );
+        return wp_sprintf( __( 'Donation Form ID: %d', 'give' ), $formId );
```
Comment (ravinderk) [inline src/Framework/PaymentGateways/DonationSummary.php:79]: @kjohnson any specific reason to use `wp_sprintf`?

File: src/Framework/PaymentGateways/DonationSummary.php [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+            $this->getSummary(),
+            $this->getDonorLabel(),
+        ]));
+    }
+
+    /**
+     * @unreleased
+     * @return string
+     */
+    public function getSummary()
```
Comment (ravinderk) [inline src/Framework/PaymentGateways/DonationSummary.php:52]: @kjohnson are we following the same template for donation summary given in `give_payment_gateway_donation_summary` function?

File: src/Framework/PaymentGateways/DonationSummary.php [hunk 3]
```diff
@@ -0,0 +1,120 @@
+<?php
+
+namespace Give\Framework\PaymentGateways;
+
+use Give\PaymentGateways\DataTransferObjects\GatewayPaymentData;
+
+/**
+ * @unreleased
+ */
+class DonationSummary
```
Comment (ravinderk) [inline src/Framework/PaymentGateways/DonationSummary.php:10]: @kjohnson which function in this class return value equivalent to `give_payment_gateway_item_title`?

File: src/Framework/PaymentGateways/DonationSummary.php (hunk 1 above)
Comment (kjohnson) [reply src/Framework/PaymentGateways/DonationSummary.php:79]: Interesting: https://andy.wordpress.com/2011/01/05/wp_sprintf/

Comment (ravinderk) [reply src/Framework/PaymentGateways/DonationSummary.php:79]: @kjohnson interesting read.

I think we can replace `wp_sprintf ` with `sprintf`.

Comment (kjohnson) [reply src/Framework/PaymentGateways/DonationSummary.php]: @ravinderk updated, see https://github.com/impress-org/givewp/pull/6209/commits/934de94a40a511032b38dc8049170dbb4eea57b1

File: src/Framework/PaymentGateways/DonationSummary.php (hunk 2 above)
Comment (kjohnson) [reply src/Framework/PaymentGateways/DonationSummary.php:52]: Yes, I test this against the `give_payment_gateway_donation_summary` function directly to ensure the same return value.

```php
$this->assertEquals(
    give_payment_gateway_donation_summary($this->get_legacy_donation_data($paymentData), false),
    $summary->getSummary()
);
```

File: src/Framework/PaymentGateways/DonationSummary.php (hunk 3 above)
Comment (kjohnson) [reply src/Framework/PaymentGateways/DonationSummary.php:10]: I'm not seeing `give_payment_gateway_item_title()` used by `give_payment_gateway_donation_summary()`. Instead, the original function checks the donation data for a passed form title.

```php
// function give_payment_gateway_donation_summary

// Form title.
$summary = ( ! empty( $donation_data['post_data']['give-form-title'] )
  ? $donation_data['post_data']['give-form-title']
  : ( ! empty( $form_id )
	  ? wp_sprintf( __( 'Donation Form ID: %d', 'give' ), $form_id )
	  : __( 'Untitled donation form', 'give' ) ) );
```

**Review by ravinderk — APPROVED** [review]
@kjohnson Good job 👍

**Pull request merged**